File: strawpool-react/app/dto/PollDto.ts
import type { PollType } from "~/components/domain/PollType"
import type { OptionDto } from "./OptionDto"
import type { UserDto } from "./UserDto"

export type PollDto = {
    id: string | null,
    title: string,
    description: string | null,
    pollType: PollType,
    options: OptionDto[],
    user: UserDto | null,
    anonymousId: string | null
}

File: strawpool-spring/src/main/java/com/strawpool_spring/dto/PollDto.java
package com.strawpool_spring.dto;

import com.strawpool_spring.domain.PollType;
import com.strawpool_spring.entity.Option;
import com.strawpool_spring.entity.User;

import java.util.List;

public record PollDto (
        String id, String title, String description, PollType pollType, List<OptionDto> options, User user,String anonymousId) {
}
